### PR TITLE
save summary stats and distance error for paths

### PR DESCRIPTION
**Pull request by efahnestock**

### CODE REVIEW

File: experimental/overhead_matching/swag/evaluation/evaluate_swag.py [hunk 1]
```diff
@@ -115,6 +144,29 @@ def run_inference_on_path(
     particle_history.append(particle_state.cpu().clone())
     return particle_history
 
+def construct_inputs_and_evalulate_path(
```
Comment (efahnestock) [inline experimental/overhead_matching/swag/evaluation/evaluate_swag.py:147]: pulling function out for use in other visualization tool

File: experimental/overhead_matching/swag/evaluation/evaluate_swag.py [hunk 2]
```diff
@@ -47,6 +49,33 @@ def evaluate_prediction_top_k(
     df = df.set_index("panorama_index", drop=True)
     return df
 
+def get_distance_error_meters(
```
Comment (ewfuentes) [inline experimental/overhead_matching/swag/evaluation/evaluate_swag.py:52]: consider a more descriptive function name.

File: experimental/overhead_matching/swag/evaluation/evaluate_swag.py [hunk 3]
```diff
@@ -156,24 +207,34 @@ def evaluate_model_on_paths(
...
-            initial_particle_state = sa.initialize_wag_particles(
-                gt_initial_position_lat_lon, wag_config, generator).to(device)
-            particle_history = run_inference_on_path(sat_patch_kdtree,
-                                                     initial_particle_state,
-                                                     motion_deltas,
-                                                     path_similarity_values,
-                                                     wag_config,
-                                                     generator)
-            # save particle history
+            generator_seed = seed * i
```
Comment (ewfuentes) [inline experimental/overhead_matching/swag/evaluation/evaluate_swag.py:210]: Is the initial seed saved somewhere? Consider saving the exact seed next to the evaluation path.

File: experimental/overhead_matching/swag/evaluation/evaluate_swag.py [hunk 4]
```diff
@@ -156,24 +207,34 @@ def evaluate_model_on_paths(
...
+            )
+
             save_path = output_path / f"{i:07d}"
             save_path.mkdir(parents=True, exist_ok=True)
             particle_history = torch.stack(particle_history)
-            particle_history = particle_history[:, ::10].contiguous()  # downsample particles
-            torch.save(particle_history, save_path / "particle_history.pt")
+            error_meters_at_each_step = torch.tensor(get_distance_error_meters(vigor_dataset, path, particle_history))
+            all_final_particle_error_meters.append(error_meters_at_each_step[-1])
+            torch.save(error_meters_at_each_step, "error.pt")
```
Comment (ewfuentes) [inline experimental/overhead_matching/swag/evaluation/evaluate_swag.py:227]:
```suggestion
            torch.save(error_meters_at_each_step, save_path / "error.pt")
```

File: experimental/overhead_matching/swag/evaluation/evaluate_swag_test.py [hunk 5]
```diff
@@ -131,6 +131,62 @@ def test_evaluate_prediction_top_k(self):
...
+            [37.7749, -122.4194],  # SF estimate
+            [34.0522, -118.2437]   # LA estimate (exact match to demonstrate issue)
+        ])
+        multi_particles = torch.zeros((2, 10, 2))
+        for i in range(2):
+            multi_particles[i] = particle_means[i].unsqueeze(0) + torch.randn(10, 2) * 0.0001
+            
+        # ISSUE: The current implementation will calculate incorrect distances
+        # when multiple panorama indices are provided because it doesn't index
+        # true_latlong and particle_latlong_estimate by i in the loop.
```
Comment (ewfuentes) [inline experimental/overhead_matching/swag/evaluation/evaluate_swag_test.py:180]: I don't think I understand the issue... Is it something about how the metadata is stored in a dataframe?

File: experimental/overhead_matching/swag/evaluation/evaluate_swag.py (hunk 3 above)
Comment (ewfuentes) [reply experimental/overhead_matching/swag/evaluation/evaluate_swag.py:210]: In a different PR.

**Review by ewfuentes — APPROVED** [review]

**Comment by ewfuentes** [comment]
lint error should go away if you rebase on main
